src/GameLogic/DataStructures/ActionSequence.ts

```diff
@@ -0,0 +1,24 @@
+import {PlayerAction} from "../Enums";
+
+export class ActionSequence {
+    private _actionSequence: PlayerAction[];
+    private _step: number = 0;
+
+	constructor(actionSequence: PlayerAction[] = [])
+	{
+		this._actionSequence = actionSequence;
+    }
+    
+    public push(action: PlayerAction): void {
+		this._actionSequence.push(action);
+    }
+    
+    public getNext(): PlayerAction {
+        if (this._step >= this._actionSequence.length) {
+            return undefined;
+        }
+        let output = this._actionSequence[this._step];
```

Comment:
`let` to `const` or alternatively just `return this._actionSequence[this._step++];`